src/compiler/checker.ts

```diff
@@ -16114,7 +16121,7 @@ namespace ts {
 
         // In an assignment expression, the right operand is contextually typed by the type of the left operand.
         // Don't do this for special property assignments unless there is a type tag on the assignment, to avoid circularity from checking the right operand.
-        function isContextSensitiveAssignment(binaryExpression: BinaryExpression): boolean {
+        function isContextSensitiveAssignment(binaryExpression: BinaryExpression): boolean | Type {

```

Comment:
Can we invent a naming convention for "predicate with sidechannel truthy value" ?

Reply:
`isXOrOverriden`?